test/uk/gov/hmrc/gform/core/TemplateValidatorSpec.scala

```diff
@@ -84,11 +85,11 @@ class TemplateValidatorSpec extends Spec {
           case expr @ HasExpr(SingleExpr(UserCtx(EnrolledIdentifier))) => expr
         }.isEmpty
 
-        whenever(
-          newTemplate.authConfig == HmrcSimpleModule || newTemplate.authConfig
-            .isInstanceOf[HmrcAgentModule] && isAUserCtx) {
-          validateEnrolmentIdentifier(newTemplate) should be(Invalid("Invalid auth type for component type."))
+        whenever(newTemplate.authConfig == HmrcSimpleModule
+          || newTemplate.authConfig.isInstanceOf[HmrcAgentModule] || newTemplate.authConfig == Anonymous && isAUserCtx) {
```

Comment:
Rather than generating a completely random FormTemplate & then constraining the conditions under which it can apply to the test, replace the parts of the random FormTemplate with what you need. e.g.

```scala
forAll(FormTemplateGen.formTemplateGen, Gen.oneOf(AuthConfigGen.hmrcAgentModuleGen, Gen.const(HmrcSimpleModule)) { (initialTemplate, authConfig) =>
  val template = initialTemplate.copy(authConfig = authConfig)
  ...
```

Not only does this remove the isInstanceOf check, but it also avoids the potential for the spec to fail because the generators tried too many times to generate a FormTemplate that didn't conform to the whenever constraint.

Reply:
makes perfect sense! Done it